__tests__/commands/install.js

```diff
@@ -204,6 +204,15 @@ test.concurrent('install should dedupe dependencies avoiding conflicts 1', (): P
   });
 });
 
+test.concurrent('check and install should verify integrity in the same way when flat', (): Promise<void> => {
+  // A@2.0.1 -> B@2.0.0
+  // should result in B@2.0.0 flattened
+  return runInstall({flat: true, integrity: true}, 'install-should-dedupe-avoiding-conflicts-1', async (config) => {
```

Comment:
runInstall has an implicit check to verify that all install operations are consistent.
It would be better to run an explicit check with integrity flag because we may refactor runInstall eventually and this test would be hard to keep consistent.